RNA_folding.py

```
@@ -0,0 +1,329 @@
+
+    # Create constraints for overlapping and and substem containment.
+    quadratic_coeffs = pseudoknot_terms(stem_dict, min_stem=min_stem)
+
+    bqm = dimod.BinaryQuadraticModel(linear_coeffs, quadratic_coeffs, 'BINARY')
+
+    cqm = dimod.ConstrainedQuadraticModel()
+    cqm.set_objective(bqm)
+
+    # Add constraint disallowing overlapping substsems included in same maximal stem.
+    for stem, substems in stem_dict.items():
+        if len(substems) > 1:
+            # Add the variable for all zeros case in one-hot constraint
+            zeros = 'Null:' + str(stem)
+            cqm.add_variable(zeros, 'BINARY')
+            cqm.add_discrete(substems + [zeros], stem)
+
+    for stem1, stem2 in combinations(stem_dict.keys(), 2):
+        # Check maximal stems first.
+        if check_overlap(stem1, stem2):
+            # If maximal stems overlap, compare list of smaller stems.
+            for stem_pair in product(stem_dict[stem1], stem_dict[stem2]):
+                if check_overlap(stem_pair[0], stem_pair[1]):
+                    cqm.add_constraint(dimod.quicksum([dimod.Binary(stem) for stem in stem_pair]) <= 1)
+
+    return cqm
+
+
+def process_cqm_solution(sample_set, verbose=True):
+    """ Processes samples from solution and prints relevant information.
+
+    Prints information about the best feasible solution and returns a list of stems contained in solution.
+    Returns solution as a list of stems rather than a binary string.
+
+    Args:
+        sample_set: dimod sampleset element.
+        verbose: Boolean indicating if function should print additional information.
+
+    Returns: List of stems included in optimal solution, encoded as 4-tuples.
+    """
+
+    # Filter for feasibility
+    feasible_samples = sample_set.filter(lambda s: s.is_feasible)
+    # Check that feasible example exists
+    if not feasible_samples:
+        print('\033[93m' + '\nWarning! All solutions infeasible. You may need to try again.')
+        return None
```

It looks like this returning `None` here leads to a later exception down the line when calling `make_plots`.  You could raise an exception here, call `sys.exit()`, or handle the `None` case in `main`.

Btw, an easy way to test the behavior is to just add `or True` to the if statement.

---

I always forget about sys.exit()